src/shared/specifier.c

```diff
@@ -301,6 +302,27 @@ int specifier_os_image_version(char specifier, const void *data, const char *roo
         return parse_os_release(root, "IMAGE_VERSION", ret);
 }
 
+int specifier_credentials_dir(char specifier, const void *data, const char *root, const void *userdata, char **ret) {
+        const char *d;
+        char *copy;
+        int r;
+
+        assert(ret);
+
+        r = get_credentials_dir(&d);
```

Comment:
IIUC, this reads values from the `CREDENTIALS_DIRECTORY` environment variable, but at the time the unit is loaded, the env var is not set. Hence, I guess this specifier always provides an empty string `""`.,,